### PR TITLE
Require Ruby 3.1 in Gemfile

### PR DESCRIPTION
**Pull request by st0012**
The `ruby-lsp` gem supports 3.0 or above, but its `rbi` development dependency requires Ruby 3.1. And because dependabot always uses the lowest required Ruby to run the job, it's always failing to find the right `rbi` version.

Hopefully this change will make dependabot use Ruby 3.1 and above instead.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -156,5 +156,8 @@ DEPENDENCIES
   syntax_tree (>= 6.1.1, < 7)
   tapioca (~> 0.13)
 
+RUBY VERSION
+   ruby 3.3.4p94
```
Comment (andyw8) [inline Gemfile.lock:160]: 🤔 will this be noisy if people have different patch levels of Ruby on their machines?

**Review by andyw8 — APPROVED** [review]
Let's try it

File: Gemfile.lock (hunk 1 above)
Comment (st0012) [reply Gemfile.lock:160]: It's the Ruby version bundler will prefer using. The `ruby` declaration in `Gemfile` though is a requirement.

**Comment by st0012** [comment]
Dependabot still [decided](https://productionresultssa5.blob.core.windows.net/actions-results/e99aa144-8b2c-42f1-b46e-365db95e7474/workflow-job-run-be2c39d2-39df-5cd1-ffd9-71ea239221a0/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-08-02T17%3A27%3A25Z&sig=yta3ug%2FlofPyjP%2FTflFr8c8pB0qnxweLcfeR509RqbY%3D&ske=2024-08-03T03%3A28%3A14Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2024-08-02T15%3A28%3A14Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2024-05-04&sp=r&spr=https&sr=b&st=2024-08-02T17%3A17%3A20Z&sv=2024-05-04) to use Ruby 3.0.6 😅 

> tapioca >= 0.11.11 requires Ruby >= 3.1.\nSo, because Gemfile depends on tapioca ~> 0.13\n  and current Ruby version is = 3.0.6,\n  version solving has failed."}
